futures-sink/src/channel_impls.rs

```diff
@@ -25,7 +25,7 @@ impl<T> Sink for Sender<T> {
 
 impl<T> Sink for UnboundedSender<T> {
     type SinkItem = T;
-    type SinkError = ChannelClosed<T>;
+    type SinkError = SendError;
```

Comment:
With this new error type, you can no longer get the value back if send failed. I don't think this will be too much of a problem in practice, plus it was kind of a pain because it would only *sometimes* be present in the error type.

Reply:
That's true, but there is `Sender::try_send` and `UnboundedSender::unbounded_send` if you need that fuctionality, while cleaning up everything else.

Reply:
Okay-- that seems reasonable to me. Thanks for the PR!